DllImportGenerator/DllImportGenerator/LanguageSupport.cs

```diff
@@ -0,0 +1,10 @@
+// Types defined to enable language support of various features
+// in the source generator.
+
+
+namespace System.Runtime.CompilerServices
+{
+    // Define IsExternalInit type to support records.
+    internal class IsExternalInit
```

Comment:
The `IsExternalInit` type is needed to use the new "init properties" feature in C# 9, which records use under the hood.

Reply:
So our source generator needs to define this type so that Roslyn will enable the feature? Is this point in time or a common approach. Any documentation would be helpful here.

Reply:
Yes. This is a common approach that has been used since .NET Framework 3.5 to enable features that don't require runtime support on downlevel runtimes (enabling LINQ query syntax on older .NET Framework versions is the first case I know of, our BCL IAsyncEnumerable package for .NET Standard 2.0 to enable `await foreach` is another example).

Reply:
I know about the BCL side of the house. I am more intrigued by an extension point though. This seems like it is/should be a point in time issue and not something we will need to have long term or does Roslyn use this model throughout its extensibility architecture?

Reply:
Since the source generator has to target .NET Standard 2.0, we'll have to keep this long term. If we were able to have the source generator target .NET 5.0, then we'd be able to remove it.

Reply:
I'm not sure why Roslyn chose to not emit the type when it isn't present for IsExternalInit since for older features they used to emit the extra types needed for metadata when they weren't present. Maybe it's because this is a modreq instead of an attribute?

Reply:
The official support matrix for C# versions and runtime versions is https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/configure-language-version#defaults . Records are C# 9 feature and so they are not officially supported for .net standard.

If you know what you are doing, you can make some of C# 9 features work using hacks like this.

Reply:
>If you know what you are doing, you can make some of C# 9 features work using hacks like this.

I think that is the surprise for me. Adding a type to enable a language feature in the code being used is a bit indirect and abstruse. Not a fan, but get it.

Reply:
The language feature is enabled by specifying C# 9 on the compiler command line (`<LangVersion>` in msbuild). This attribute is not enabling the language feature. It is "making the language feature work" in officially unsupported configuration.